tests/buffer/buffer_storage_common.h

```diff
@@ -63,21 +63,19 @@ class buffer_storage_test {
 
     // Case 3 - Weak pointer
     std::shared_ptr<T[]> data_shared_ptr(new T[size]);
-    std::weak_ptr<T[]> data_final3 = data_shared_ptr;
+    std::shared_ptr<T> data_shared_ptr_alias(data_shared_ptr,
```

Comment:
Why doing this, since you initialize it also with the pointer from the shared pointer handling the ownership? :thinking: 
OK, after more thoughts you do this to have the same shared pointer but returning a pointer of the array decay.
You could add a comment. :smile:

Reply:
Added a comment!